apps/server/src/modules/room-member/service/room-member.service.ts

```diff
@@ -0,0 +1,82 @@
+import { EntityManager, ObjectId } from '@mikro-orm/mongodb';
+import { Injectable } from '@nestjs/common';
+import { User } from '@shared/domain/entity';
+import { RoleName } from '@shared/domain/interface';
+import { EntityId } from '@shared/domain/types';
+import { RoleRepo } from '@shared/repo';
+import { Action, AuthorizationService } from '@src/modules/authorization';
+import { Group, GroupService, GroupTypes } from '@src/modules/group';
+import { GroupEntity, GroupEntityTypes, GroupUserEmbeddable } from '@src/modules/group/entity';
+import { RoomMemberEntity } from '../repo/entity/room-member.entity';
+import { RoomMemberRepo } from '../repo/room-member.repo';
+import { roomMemberEntityFactory } from '../testing';
+
+@Injectable()
+export class RoomMemberService {
+	constructor(
+		private readonly roomMembersRepo: RoomMemberRepo,
+		private readonly groupService: GroupService,
+		private readonly roleRepo: RoleRepo,
+		private readonly em: EntityManager,
+		private readonly authorizationService: AuthorizationService
+	) {}
+
+	private async createNewRoomMemberWithEditorRole(roomId: EntityId, groupUser: GroupUserEmbeddable) {
+		const newGroup = new Group({
+			name: `Room Members for Room ${roomId}`,
+			externalSource: undefined,
+			type: GroupTypes.OTHER,
+			users: [{ userId: groupUser.user.id, roleId: groupUser.role.id }],
+			id: new ObjectId().toHexString(),
+		});
+		const savedGroup = await this.groupService.save(newGroup);
+		const groupEntity = new GroupEntity({
```

Comment:
thinking this through, im not sure what the ideal solution is. I see two possible approaches.

The first one, is to use an id instead of a reference in the DO, and simply use it as an inbetween step to fetching the group next. That is definetly clean dependencywise, but also feels pretty cluncy.

The second approach is to always have the group populated in the RoomMembers object, and pushing the responsability for mapping this into the repository (since the repository should create Domain Objects).
Again, the goal should be to not knowing about the GroupEntity or the GroupDomainMapper. The cleanest would be to have only the id in the entity, and after loading the entity calling the group service from the Repo to fetch the DO for the Group, and then creating the GroupMembers DO with it. The disadvantage here is that we need two calls to the database.

a more "pragmatic" approach would be to break some rules, and simply import the things you need to do the mapping in the RoomMembers Repo. At least it would retain the borders and responsabilities of the layers, though it would mean a very close coupling to the group module. (and a break of our import rules)

An in-between solution could be to still do the populate in the database command, but without knowing what it means, and then fetching the DO from the other module via its id, relying on the identitymap to avoid another call to the database. But I dont know exactly if and how that could work, as I think the populate requires at least the entity to know the other entity, thus already breaking module borders... so I think this one seems unfeasable for now.

For this PR, im fine with either solution, as long as layer responsabilities are kept. But for the future we should figure out how to deal with such a case, where "id only" might not suffice as a hard rule.